Bugs/7593 - with test fix

bradhead opened the pull request:
Added authZ checks to Immz and Patient Service and tidyed up the service result to use Forbid on authZ failure and construct a new JsonResult()

TiagoGraf89 commented on Apps/Immunization/src/Controllers/ImmunizationController.cs:
```diff
@@ -35,27 +39,50 @@ public class ImmunizationController : ControllerBase
+        /// <param name="hdid">The health directed id for the subject.</param>
         /// <returns>a list of immunization records.</returns>
         /// <response code="200">Returns the List of Immunization records.</response>
         /// <response code="401">The client is not authorzied to retrieve the list.</response>
```
nit: 403 is missing

healthgatewaybc commented:
Patient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=patientservice_pr_183)

healthgatewaybc commented:
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_183)

healthgatewaybc commented:
Patient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=patientservice_pr_183)

sslaws commented:
Immunization tests are failing... please fix issue before merging.

healthgatewaybc commented:
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_183)

healthgatewaybc commented:
Immunization Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=immunization_pr_183)

healthgatewaybc commented:
Patient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=patientservice_pr_183)

healthgatewaybc commented:
Immunization Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=immunization_pr_183)

healthgatewaybc commented:
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_183)

healthgatewaybc commented:
Patient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=patientservice_pr_183)

bradhead commented:
> Imms tests are failing

the darn functional tests are old thinking.. hard-coded base Url, and the overall route wasn't thought out. I fixed the test for now... so we can assert that a 401 works now, but this needs to be looked at once we resume Immz. I just wanted to be sure we have Authorization protections on all of our services, even this one... updated this evening.
